### PR TITLE
Control other particles from the Sodium Extra GUI;

### PR DESCRIPTION
**Pull request by HarvelsX**
Players have the need to disable particles of other modifications, but unfortunately the implementation of this happens through many other interfaces (_ModMenu_, etc.), which makes it very difficult to easily access this functionality.

This PR adds particles of other modifications to the GUI, and was created as a common solution for this PR: https://github.com/RandomMcSomethin/fallingleaves/pull/34

### CODE REVIEW

**Comment by FlashyReese** [comment]
Since this is using the particle registry to generate toggles, I assume we can get rid of some of the existing particles toggles.

**Comment by HarvelsX** [comment]
Yes, we can, but it doesn't group the particles.

File: src/main/java/me/flashyreese/mods/sodiumextra/client/gui/SodiumExtraGameOptionPages.java [hunk 1]
```diff
@@ -208,6 +212,22 @@ public static OptionPage particle() {
...
+                .collect(Collectors.groupingBy(Identifier::getNamespace));
+        otherParticles.forEach((namespace, identifiers) -> groups.add(identifiers.stream().collect(
+                OptionGroup::createBuilder,
+                (builder, identifier) -> builder.add(
+                        OptionImpl.createBuilder(boolean.class, sodiumExtraOpts)
+                                .setName(Text.translatable(identifier.toTranslationKey("options.particles")))
+                                .setTooltip(Text.translatable(identifier.toTranslationKey("options.particles").concat(".tooltip")))
+                                .setControl(TickBoxControl::new)
+                                .setBinding((opts, val) -> opts.particleSettings.otherMap.put(identifier, val),
+                                        opts -> opts.particleSettings.otherMap.get(identifier))
```
Comment (FlashyReese) [inline src/main/java/me/flashyreese/mods/sodiumextra/client/gui/SodiumExtraGameOptionPages.java]: Due to how the configuration is implemented, I think the workaround fix for this is to dynamically set the value for them.

Edit: This isn't the most elegant solution, but it works and can load new mods with particle changes that's not in the config.
```suggestion
                                        opts ->  {
                                            if (!opts.particleSettings.otherMap.containsKey(identifier))
                                                opts.particleSettings.otherMap.put(identifier, true);

                                            return opts.particleSettings.otherMap.get(identifier);
                                        }
```

**Comment by FlashyReese** [comment]
> Yes, we can, but it doesn't group the particles.

This gives the user more fine control than grouping imo. Thoughts on just removing the groups and letting users toggle with fine detail?

**Comment by HarvelsX** [comment]
> This gives the user more fine control than grouping imo. Thoughts on just removing the groups and letting users toggle with fine detail?

Okay, but I'll still look at the grouping options

**Comment by FlashyReese** [comment]
LGTM

**Pull request merged**